src/NuGet.Core/NuGet.Commands/RestoreCommand/RestoreRequest.cs

```diff
@@ -86,7 +89,7 @@ public class RestoreRequest
         /// <see cref="DefaultDegreeOfConcurrency" />. Set this to '1' to
         /// run without concurrency.
         /// </summary>
-        public int MaxDegreeOfConcurrency { get; set; } = DefaultDegreeOfConcurrency;
+        public int MaxDegreeOfConcurrency { get; set; }
```

Comment:
This value specifically refers to the number of concurrent threads running package installation. 
https://github.com/NuGet/NuGet.Client/blob/31bb76ab2c41d2450927a42de356fcbdfc9d3672/src/NuGet.Core/NuGet.Commands/RestoreCommand/ProjectRestoreCommand.cs#L236

While this does seem like the right direction, I think we might need some consolidation of all the concurrency switches. 

Furthermore, we'd need performance numbers on how this impacts the performance on more powerful machines. 

//cc @dtivel @rrelyea